internal/integration/integration_test.go

```diff
@@ -102,6 +102,33 @@ func (suite *SpokesReceivePackTestSuite) TestSpokesReceivePackMultiplePush() {
 		"unexpected error running the push with the custom spokes-receive-pack program")
 }
 
+func (suite *SpokesReceivePackTestSuite) TestSpokesReceivePackMultiplePushWithExtraReceiveOptions() {
+	assert.NoError(suite.T(), os.Chdir(suite.remoteRepo), "unable to chdir into our remote Git repo")
+	require.NoError(suite.T(), exec.Command("git", "config", "receive.fsckObjects", "true").Run())
+	// This value is the default value we set in our production config
+	require.NoError(suite.T(), exec.Command("git", "config", "receive.maxsize", "2147483648").Run())
+
+	assert.NoError(suite.T(), os.Chdir(suite.localRepo), "unable to chdir into our local Git repo")
+	assert.NoError(
+		suite.T(),
+		exec.Command(
+			"git", "push", "--all", "--receive-pack=spokes-receive-pack-wrapper", "r").Run(),
+		"unexpected error running the push with the custom spokes-receive-pack program")
+}
+
+func (suite *SpokesReceivePackTestSuite) TestSpokesReceivePackMultiplePushFailMaxSize() {
```

Comment:
Is it possible to have a test for the fsck config as well?  Also do we want to assert on the right error message as well here, instead of just the push failing?

Reply:
You mean an additional one to `TestSpokesReceivePackMultiplePushWithExtraReceiveOptions`?

In this test I have already set `require.NoError(suite.T(), exec.Command("git", "config", "receive.fsckObjects", "true").Run())`

Reply:
In have included here 28f9a70b263123e54ac22f48cfd369d873094bc6 a more exhaustive validation

Reply:
I meant another test that actually rejects a push because of the `receive.fsckObjects` config.  That might require handcrafting a broken object we can send since the regular git client will probably reject it.  The current config just checks that nothing is failing with a regular push, but doesn't actually check if the fsck works correctly.

> In have included here https://github.com/github/spokes-receive-pack/commit/28f9a70b263123e54ac22f48cfd369d873094bc6 a more exhaustive validation

That looks like great additional validation for `receive.maxSize`!

Reply:
Makes sense, I will need to figure out how to handcraft a broken object :D

Reply:
I have pushed the additional test here ec479b8538c74a0c3317f433638cd91b187d4d47 Thanks a lot for the hint about how to create a bogus object!